packages/contracts/contracts/plugin/psp/PluginSetupProcessor.sol

```diff
@@ -0,0 +1,703 @@
+            revert PluginAlreadyInstalled();
+        }
+
+        // If the plugin block number exceeds the setupId preparation block number,
+        // This means applyInstallation was already called on another setupId
+        // and all the rest setupIds should become idle or setupId is not prepared before.
+        if (pluginInformation.blockNumber >= pluginInformation.setupIds[setupId]) {
+            revert SetupNotEligible(setupId);
+        }
+
+        bytes32 newSetupId = _getSetupId(
+            _params.pluginSetupRef,
+            bytes32(0),
+            _params.helpersHash,
+            bytes32(0),
+            bytes("")
+        );
+
+        pluginInformation.currentSetupId = newSetupId;
+        pluginInformation.blockNumber = block.number;
+
+        _executeOnDAO(_dao, setupId, _params.permissions, _params.actions);
+
+        emit InstallationApplied({dao: _dao, plugin: _params.plugin, setupId: setupId});
+    }
+
+    /// @notice Prepares the update of an UUPS upgradeable plugin.
+    /// @param _dao The address of the installing DAO.
+    /// @param _params The struct containing the parameters for the `prepareUpdate` function.
+    /// @return initData The initialization data to be passed to upgradeable contracts when the update is applied
+    /// @return preparedDependency TOD:GIORGI
+    /// @dev The list of `_currentHelpers` has to be specified in the same order as they were returned from previous setups preparation steps (the latest `prepareInstallation` or `prepareUpdate` step that has happend) on which the update is prepared for
+    function prepareUpdate(address _dao, PrepareUpdate calldata _params)
+        external
+        returns (bytes memory initData, IPluginSetup.PreparedDependency memory preparedDependency)
+    {
+        if (
+            _params.currentVersionTag.release != _params.newVersionTag.release ||
+            _params.currentVersionTag.build <= _params.newVersionTag.build
+        ) {
+            revert UpdateVersionInvalid({
+                currentVersionTag: _params.currentVersionTag,
+                newVersionTag: _params.newVersionTag
+            });
+        }
+
+        // // Check that plugin is `PluginUUPSUpgradable`.
+        if (!_params.setupPayload.plugin.supportsInterface(type(IPlugin).interfaceId)) {
+            revert IPluginNotSupported({plugin: _params.setupPayload.plugin});
+        }
+        if (IPlugin(_params.setupPayload.plugin).pluginType() != IPlugin.PluginType.UUPS) {
+            revert PluginNonupgradeable({plugin: _params.setupPayload.plugin});
+        }
+
+        PluginRepo.Version memory currentVersion = _params.pluginSetupRepo.getVersion(
+            _params.currentVersionTag
+        );
+
+        PluginRepo.Version memory newVersion = _params.pluginSetupRepo.getVersion(
+            _params.newVersionTag
+        );
+
+        if (currentVersion.pluginSetup == newVersion.pluginSetup) {
+            // revert plugin setups can't be the same(pointless)
```

Comment:
missing error.

Reply:
I was waiting for another change to go there, so, now it has. no need for the revert anymore.